content/blog/gsoc-25-netjsongraph.rst

```diff
@@ -0,0 +1,225 @@
+
+Summary of Contributions & Links to Work
+----------------------------------------
+
+I successfully implemented the following key features and fixes through a series of pull requests:
+
+1.  **Feature: Prevent Overlapping of Clusters** — Developed a sophisticated algorithm to visually separate data clusters that share the same geographic location but belong to different categories, solving a major data visibility issue.
+
+    - **Pull Requests:** `#396 [netjsongraph.js] <https://github.com/openwisp/netjsongraph.js/pull/396>`_, `#349 [netjsongraph.js] <https://github.com/openwisp/netjsongraph.js/pull/349>`_, `#668 [openwisp-monitoring] <https://github.com/openwisp/openwisp-monitoring/pull/668>`_
+
+2.  **Feature: Display Connected Clients on Nodes** — Implemented a flexible, modular overlay to display the number of connected clients as dots orbiting each node, with a highly efficient placement algorithm.
+
+    - **Pull Request:** `#411 [netjsongraph.js] <https://github.com/openwisp/netjsongraph.js/pull/411>`_
+
+3.  **Feature: Zoom-Dependent Label Visibility** — Added options to show node labels only when the user zooms past a certain threshold, decluttering the view for both map and graph modes.
+
+    - **Pull Requests:** `#407 [map] <https://github.com/openwisp/netjsongraph.js/pull/407>`_, `#419 [graph] <https://github.com/openwisp/netjsongraph.js/pull/419>`_
+
+4.  **Fix: Resiliency for Invalid Data** — Hardened the library against data with duplicate node IDs. It now handles this issue gracefully by deduplicating nodes and logging a warning instead of crashing.
+
+    - **Pull Request:** `#355 [netjsongraph.js] <https://github.com/openwisp/netjsongraph.js/pull/355>`_
+
+5.  **Fix: Respect Map Tile Provider Zoom Levels** — Fixed the map's zoom behavior to honor the ``minZoom`` and ``maxZoom`` limits of the active tile provider, improving the user experience with visual cues on the zoom controls.
+
+    - **Pull Request:** `#363 [netjsongraph.js] <https://github.com/openwisp/netjsongraph.js/pull/363>`_
+
+Detailed Technical Breakdown of Work
+------------------------------------
+
+Core Achievement: Advanced Cluster Overlap Prevention
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This was the main goal of the project, designed to solve a critical issue where overlapping clusters would hide important information on the map.
+
+- **The Problem:** When multiple nodes are located at the same geographic coordinates, the library groups them into a single cluster. However, if these nodes belonged to different categories (e.g., status "ok" vs. "critical"), the resulting clusters would be rendered directly on top of each other, making it impossible to see the status breakdown.
```

Comment:
This subdivision using lists doesn't look great, try to use subheadings to divide this section in subsections